src/fight/mod.rs

```diff
@@ -0,0 +1,1058 @@
+        / global_values::DEFAULT_DICE) as f64)
+        .ceil() as usize;
+    (((nb_hit * (global_values::DEFAULT_DICE - to_wound + 1)) / global_values::DEFAULT_DICE) as f64)
+        .ceil() as usize
+}
+
+fn get_final_result(
+    nb_wounds: usize,
+    regiment_defending_armor: usize,
+    regiment_attacking_armor_penetration: usize,
+) -> usize {
+    let to_save: i8 = regiment_defending_armor as i8 - regiment_attacking_armor_penetration as i8;
+    let throw_to_save: i8 = match to_save {
+        i8::MIN..=0 => return nb_wounds,
+        1 => 6,
+        2 => 5,
+        3 => 4,
+        4 => 3,
+        5..=i8::MAX => 2,
+    };
+    let nb_save: usize = ((nb_wounds) * (global_values::DEFAULT_DICE - throw_to_save as usize + 1))
+        / global_values::DEFAULT_DICE;
+    nb_wounds - nb_save
+}
+
+fn fight_first_turn(
+    regiment_attacking: &regiment::Regiment,
+    regiment_defending: &regiment::Regiment,
+) -> usize {
+    let regiment_attacking_stats: &model::Stats = regiment_attacking.get_model().get_stats();
+    let regiment_defending_stats: &model::Stats = regiment_defending.get_model().get_stats();
+    let to_hit: usize = roll::compute_roll_to_hit(
+        regiment_attacking_stats.get_offensive(),
+        regiment_defending_stats.get_defense(),
+    );
+    let to_wound: usize = roll::compute_roll_to_wound(
+        regiment_attacking_stats.get_strength(),
+        regiment_defending_stats.get_resilience(),
+    );
+    let nb_attacks: usize = get_nb_attacks(regiment_attacking);
+    let nb_wounds: usize = get_nb_wounds(nb_attacks, to_hit, to_wound);
+    let final_result: usize = get_final_result(
+        nb_wounds,
+        regiment_defending_stats.get_armour(),
+        regiment_attacking_stats.get_armour_penetration(),
+    );
+    final_result
+}
+
+fn find_the_fastest(model_1: &model::Model, model_2: &model::Model) -> usize {
+    let model_1_agility: usize = model_1.get_stats().get_agility();
+    let model_2_agility: usize = model_2.get_stats().get_agility();
+
+    match model_1_agility.cmp(&model_2_agility) {
+        std::cmp::Ordering::Greater => 1,
+        std::cmp::Ordering::Less => 2,
+        std::cmp::Ordering::Equal => 0,
+    }
+}
+
+pub fn resolve_fight(regiment_1: regiment::Regiment, regiment_2: regiment::Regiment) -> usize {
+    let fastest: usize = find_the_fastest(regiment_1.get_model(), regiment_2.get_model());
+
+    match fastest {
+        1 => fight_first_turn(&regiment_1, &regiment_2),
+        2 => fight_first_turn(&regiment_2, &regiment_1),
+        0 => unimplemented!("They will fight at the same time !"),
+        _ => unreachable!("This should never happen"),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{model, regiment, roll};
+
+    use super::{
+        fight_first_turn, find_the_fastest, get_final_result, get_nb_attacks, get_nb_wounds,
+        resolve_fight,
+    };
+
+    fn initialize_chaos_warrior() -> regiment::Regiment {
```

Comment:
we can do it later and not in this PR, but could be a great idea to have a mini lib for unit test setup like this one no ?
What do you think ?

Reply:
I think we can create something like this but maybe in another sprint.

Reply:
Let's keep it in mind for a next meeting.